packages/product-json-to-csv/test/main.spec.js

```diff
@@ -0,0 +1,625 @@
+        state: { id: 'state-id' },
+        categories: [{ id: 'cat-id-1' }, { id: 'cat-id-2' }],
+        categoryOrderHints: { 'cat-id-1': '0.012', 'cat-id-2': '0.987' },
+      }
+    })
+
+    describe('::_resolveReferences', () => {
+      beforeEach(() => {
+        const productType = {
+          id: 'product-type-id',
+          name: 'resolved-product-type',
+          attributes: [{}],
+        }
+        const taxCategory = {
+          id: 'tax-cat-id',
+          key: 'resolved-tax-cat-key',
+        }
+        const state = {
+          id: 'state-id',
+          key: 'res-state-key',
+        }
+        const categories = [
+          { id: 'cat-id-1', name: { en: 'res-cat-name-1' } },
+          { id: 'cat-id-2', name: { en: 'res-cat-name-2' } },
+        ]
+        const categoryOrderHints = {
+          'res-cat-name-1': '0.015',
+          'res-cat-name-2': '0.987',
+        }
+
+        productJsonToCsv._resolveProductType = jest.fn(() => ({ productType }))
+        productJsonToCsv._resolveTaxCategory = jest.fn(() => ({ taxCategory }))
+        productJsonToCsv._resolveState = jest.fn(() => ({ state }))
+        productJsonToCsv._resolveCategories = jest.fn(() => ({ categories }))
+        productJsonToCsv._resolveCategoryOrderHints = jest.fn(() => ({
+          categoryOrderHints,
+        }))
+      })
+
+      it('should pass the products to all resolver functions', async () => {
+        await productJsonToCsv._resolveReferences(sampleProduct)
+
+        expect(productJsonToCsv._resolveProductType).toBeCalledWith(
+          sampleProduct.productType
+        )
+        expect(productJsonToCsv._resolveTaxCategory).toBeCalledWith(
+          sampleProduct.taxCategory
+        )
+        expect(productJsonToCsv._resolveState).toBeCalledWith(
+          sampleProduct.state
+        )
+        expect(productJsonToCsv._resolveCategories).toBeCalledWith(
+          sampleProduct.categories
+        )
+        expect(productJsonToCsv._resolveCategoryOrderHints).toBeCalledWith(
+          sampleProduct.categoryOrderHints
+        )
+      })
+
+      it('should return object with resolved references', async () => {
+        const expected = {
+          id: 'myProduct-1',
+          productType: {
+            id: 'product-type-id',
+            name: 'resolved-product-type',
+            attributes: [{}],
+          },
+          taxCategory: {
+            id: 'tax-cat-id',
+            key: 'resolved-tax-cat-key',
+          },
+          state: {
+            id: 'state-id',
+            key: 'res-state-key',
+          },
+          categories: [
+            { id: 'cat-id-1', name: { en: 'res-cat-name-1' } },
+            { id: 'cat-id-2', name: { en: 'res-cat-name-2' } },
+          ],
+          categoryOrderHints: {
+            'res-cat-name-1': '0.015',
+            'res-cat-name-2': '0.987',
+          },
+        }
+        const actual = await productJsonToCsv._resolveReferences(sampleProduct)
```

Comment:
We can up our game here with dealing with promises. An example would be

await expect(user.getUserName(5)).resolves.toEqual(bar)